### PR TITLE
Move snapshot-tool S3 and GCS tests to submodules

### PR DESCRIPTION
**Pull request by andrershov**
Currently, snapshot tool tests for S3 and GCS reside in the
`build.gradle` file.
And there is no way to run GCS tests without running S3 tests and vice
versa.
Of course, it's possible to create separate test tasks for GCS and S3,
but fixtures plugin is implemented in such a way, that it surrounds all
test tasks with fixture start/stop. 
So even if you were to run just GCS tests, Minio docker container would
be started (and not used).

The solution @mark-vieira suggested is to refactor S3 and GCS tests
into separate submodules, similar to what we're doing for
`repository-gcs` plugin.
This commit creates 2 new submodules - `:qa:s3` and
`:qa:google-cloud-storage`. 
And now it's easy to run GCS and S3 tests separately.
An additional benefit is that we now have 3 separate shorter/cleaner
build.gradle files.

`:x-pack:snapshot-tool:check` still invokes S3 and GCS tests.

### CODE REVIEW

File: x-pack/snapshot-tool/build.gradle [hunk 1]
```diff
@@ -79,134 +77,9 @@ dependencyLicenses {
   mapping from: /proto-google.*/, to: 'proto-google'
 }
 
+// There is only AbstractCleanupTests class in test directory and it's abstract
 test.enabled = false
-
```
Comment (andrershov) [inline x-pack/snapshot-tool/build.gradle]: Removed GCS and S3 related things

File: x-pack/snapshot-tool/qa/google-cloud-storage/build.gradle [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+dependencies {
+    compile project(":plugins:repository-gcs")
+    testCompile project(":test:framework")
+    testCompile project(':x-pack:snapshot-tool')
+    testCompile files(project(':x-pack:snapshot-tool').sourceSets.test.output)
+}
+
+test.enabled = false
+
+task gcsThirdPartyTests {
```
Comment (andrershov) [inline x-pack/snapshot-tool/qa/google-cloud-storage/build.gradle]: Lines below are mostly copied from the parent build.gradle GCS section

File: x-pack/snapshot-tool/qa/s3/build.gradle [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+dependencies {
+    compile project(":plugins:repository-s3")
+    testCompile project(":test:framework")
+    testCompile project(':x-pack:snapshot-tool')
+    testCompile files(project(':x-pack:snapshot-tool').sourceSets.test.output)
+}
+
+test.enabled = false
+
+task s3ThirdPartyTests {
```
Comment (andrershov) [inline x-pack/snapshot-tool/qa/s3/build.gradle]: Lines below are mostly copied from the parent build.gradle S3 section

File: x-pack/snapshot-tool/qa/build.gradle [hunk 4]
```diff
@@ -0,0 +1 @@
+group = "${group}.plugins.snapshot-tool.qa"
```
Comment (mark-vieira) [inline x-pack/snapshot-tool/qa/build.gradle]: I don't really think this is necessary. We've never going to publish these subprojects. We can ditch this entire file. Gradle projects do not require a build file.

**Review by mark-vieira — APPROVED** [review]
Minor comments and otherwise LGTM. Thanks for tackling this. This should make it much more convenient to run these test suites individually.

File: x-pack/snapshot-tool/qa/build.gradle (hunk 4 above)
Comment (andrershov) [reply x-pack/snapshot-tool/qa/build.gradle]: With this file removed I get the following error:
```
Could not determine the dependencies of task ':x-pack:snapshot-tool:check'.
> Task with path 'qa:s3:check' not found in project ':x-pack:snapshot-tool'
```

**Pull request merged**